app/models/repository/git_repositories.rb

```diff
@@ -128,6 +131,22 @@ def create_version(ontology, commit_oid, ontology_version_options)
     version
   end
 
+  def commit_for!(commit_oid)
+    instance = Commit.where(repository_id: self, commit_oid: commit_oid).
+      first_or_initialize
+    if !instance.persisted?
```

Comment:
We agreed to use `if` if the body spans over more than three lines, so this is okay. However, see the suggestion above.